roles/network_plugin/calico/tasks/install.yml

```diff
@@ -491,13 +491,20 @@
     - peer_with_router|default(false)
     - inventory_hostname == groups['kube_control_plane'][0]
 
-- name: Calico | Configure ipam strictaffinity
-  command:
-    cmd: "{{ bin_dir }}/calicoctl.sh ipam configure --strictaffinity={{ calico_ipam_strictaffinity }}"
-  register: output
-  retries: 4
-  until: output.rc == 0
-  delay: "{{ retry_stagger | random + 3 }}"
+- name: Calico | Create Calico ipam manifests
+  template:
+    src: "{{ item.file }}.j2"
+    dest: "{{ kube_config_dir }}/{{ item.file }}"
+    mode: 0644
+  with_items:
+    - {name: calico, file: calico-ipamconfig.yml, type: ipam}
+  when:
+    - inventory_hostname in groups['kube_control_plane']
+
+- name: Calico | Create ipamconfig resources
```

Comment:
The way the calico role is written is to create objects using `calicoctl apply -f -` with the contents provided in `stdin`, additionally I started pushing logic to prevent overriding user specified settings that kubespray does not know about (see https://github.com/kubernetes-sigs/kubespray/pull/8815), I think it makes sense to add this in this PR as well.

Reply:
ok, but `calicoctl apply` dose not support `ipamconfig`, https://projectcalico.docs.tigera.io/reference/calicoctl/apply

Reply:
Hmm, quite the inconsistency we have here but you are right in that there is no clean way of addressing this.